client/src/layers/TextLayer.ts

```diff
@@ -11,32 +11,70 @@ export interface TextData {
   y: number;
   offsetY?: number;
   offsetX?: number;
+  currentPair?: boolean;
 }
 
 export type FormatTextRow = (track: FrameDataTrack) => TextData[] | null;
 
 interface TextLayerParams {
   formatter?: FormatTextRow;
 }
-
 /**
+ * @param track - standard frameDataTrack info
+ * @param additionalNum Number of additional pairs to display, top pair is the current pair
+ * followed by more in descending order.
  * @returns value or null.  null indicates that the text should not be displayed.
  */
-function defaultFormatter(track: FrameDataTrack): TextData[] | null {
+function defaultFormatter(track: FrameDataTrack, additionalNum = 0): TextData[] | null {
   if (track.features && track.features.bounds) {
     const { bounds } = track.features;
+
     if (bounds && track.confidencePairs !== null) {
-      const type = track.confidencePairs[0];
-      const confidence = track.confidencePairs[1];
-      return [{
+      const lineHeight = 25;
+      const maxPairs = Math.min(track.confidencePairs.length, additionalNum);
+      let currentHeight = bounds[1] - (lineHeight * (maxPairs + 1));
+      const arr: TextData[] = [];
+      const baseType = track.trackType ? track.trackType[0] : 'unknown';
+      const baseConfidence = track.trackType ? track.trackType[1] : 1.0;
+      const currentTypeIndication = maxPairs > 0 ? '**' : '';
+      arr.push({
         selected: track.selected,
         editing: track.editing,
-        type,
-        confidence,
-        text: `${type}: ${confidence.toFixed(2)}`,
+        type: baseType,
+        confidence: baseConfidence,
+        text: `${currentTypeIndication}${baseType}: ${baseConfidence.toFixed(2)}`,
         x: bounds[2],
         y: bounds[1],
-      }];
+        currentPair: true,
+      });
+      //Now we display any additional types besides the default type
+      currentHeight += lineHeight;
+      let pairCount = 0;
+      if (track.confidencePairs.length) {
+        for (let i = 0; i < track.confidencePairs.length; i += 1) {
+          if (pairCount >= maxPairs) {
```

Comment:
If you change `maxPairs`, this line has to change, but I think it should because this isn't what you'd expect if `maxPairs` wasn't already off-by-one.